nion/swift/model/DisplayItem.py

```diff
@@ -1998,11 +1999,17 @@ def append_display_data_channel(self, display_data_channel: DisplayDataChannel,
             self.__add_display_layer_auto(display_layer, display_data_channel)
 
     def __get_unique_display_layer_color(self) -> str:
-        existing_colors = {display_layer.fill_color for display_layer in self.display_layers}
-        for color in ('#1E90FF', "#F00", "#0F0", "#00F", "#FF0", "#0FF", "#F0F", "#888", "#800", "#080", "#008", "#CCC", "#880", "#088", "#808", "#964B00"):
-            if not color in existing_colors:
-                return color
-        return '#1E90FF'
+        existing_colors = [display_layer.fill_color for display_layer in self.display_layers]
+        unused_colors = list({*Color.svg_color_map.values()})
+        matches = [False] * len(unused_colors)
+        for color_str in existing_colors:
+            source_color = Color.Color(color_str)
+            for index in range(len(matches)):
+                matches[index] |= source_color.matches_without_alpha(Color.Color(unused_colors[index]))
+        try:
+            return unused_colors[matches.index(False)]
+        except ValueError:
```

Comment:
As a side note -- I don't like using exceptions for regular code flow since it makes debugging more difficult. However, if you use the simplified algorithm above it won't be necessary anyway.